code/script.js

```diff
@@ -1,51 +1,285 @@
-  chat.scrollTop = chat.scrollHeight;
+  setTimeout (() => chat.scrollTop = chat.scrollHeight, 1)
 }
 
-// Starts here
+//Question 1: Bot greets and asks for the user's name
 const greeting = () => {
-  // here we call the function showMessage, that we declared earlier with the argument "Hello there, What's your name?" for message, and the argument "bot" for sender
-  showMessage("Hello there, What's your name?", 'bot');
-  // Just to check it out, change 'bot' to 'user' here 👆
+  showMessage(`Hello there, what's your name?`, 'bot')
+}
+
+const handleResponse = () => {
+  let userName = document.getElementById('name-input').value;
+  showMessage(`Nice to meet you ${userName}!`, 'bot') 
+  inputWrapper.innerHTML =''
+}
+
+//Question 2: Bot asks if user is having trouble choosing a movie
+const watchMovie = () => {
+  showMessage(`I'm guessing you're having trouble finding something to watch on Netflix?`, 'bot')
+  setTimeout (() => yesOrNo(), 1000)
+}
+
+//Question 2.1 Yes and No-buttons pop up, and the user should click on one or the other
+const yesOrNo = () => {
+  inputWrapper.innerHTML = `
+  <button id="option1">Yes</button>
+  <button id="option2">No</button>  
+  `
+  
+  document.getElementById("option1").addEventListener("click", () => {
+    showMessage(`Yes, I can't decide what to watch, help me!`, 'user')  
+    inputWrapper.innerHTML = ''
+    setTimeout (() => doYouNeedHelp('option1'), 1000)
+  })
+  
+  document.getElementById("option2").addEventListener("click", () => {
+    showMessage(`No thanks, I just wanted to try the functions of this bot!`, 'user')
+    inputWrapper.innerHTML = ''
+    setTimeout (() => doYouNeedHelp('option2'), 1000)
+  })
+}
+
+//Question 3: Bot's answer depends on if user chooses Yes ( if (selection === 'option1') ) or No ( else if (selection === 'option2') ) in Question 2
+const doYouNeedHelp = (selection) => {
+  if (selection === 'option1') {
+    showMessage(`I'm happy to help! How are you feeling today?`, 'bot')
+    setTimeout (() => moodChoice (), 1000)
+    
+   const moodChoice = () =>  {
+    inputWrapper.innerHTML= 
+    `<button id="optHappy">😄</button>
+    <button id="optSad">😥</button>
+    <button id="optAngry">😡</button>
+    `
+   
+    document.getElementById("optHappy").addEventListener('click', () => {
+      showMessage(`I'm feeling happy today`, 'user')
+      inputWrapper.innerHTML = ''
+      setTimeout (() => chooseByMood('optHappy'), 1000)
+    })
+
+    document.getElementById("optSad").addEventListener('click', () => {
+      showMessage(`I'm feeling sad today`, 'user')
+      inputWrapper.innerHTML = ''
+      setTimeout (() => chooseByMood('optSad'), 1000)
+    })
+
+    document.getElementById("optAngry").addEventListener('click', () => {
+      showMessage(`I'm feeling angry today`, 'user')
+      inputWrapper.innerHTML = ''
+      setTimeout (() => chooseByMood('optAngry'), 1000)
+    })
```

Comment:
Perhaps having the moodChoice function defined outside the doYouNeedHelp() function would make the if-else statements easier to follow?
Or maybe it's a matter of taste?

Reply:
Sure!